tests/core/test_variable.py

```diff
@@ -0,0 +1,558 @@
+    """Mock dependent variable to test relentless.DependentVariable"""
+    def __init__(self, *a, **b):
+        super().__init__(*a, **b)
+
+    @property
+    def value(self):
+        return np.sum([v.value for p,v in self.depends])
+
+    def _derivative(self, param):
+        if param in self.params:
+            if getattr(self, param).value > 5.0:
+                return 0.5
+            else:
+                return -0.5
+        else:
+            raise ValueError('Unknown parameter')
+
+class test_DependentVariable(unittest.TestCase):
+    """Unit tests for relentless.DependentVariable"""
+
+    def test_init(self):
+        """Test creation with data."""
+        t = relentless.DesignVariable(value=1.0)
+        u = relentless.DesignVariable(value=2.0)
+        v = relentless.DesignVariable(value=3.0)
+
+        #test creation with only vardicts
+        w = DepVar({'t':t, 'u':u, 'v':v})
+        self.assertCountEqual(w.params, ('t','u','v'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v})
+        self.assertAlmostEqual(w.value, 6.0)
+
+        #test creation with only kwvars
+        self.assertCountEqual(w.params, ('t','u','v'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v})
+        self.assertAlmostEqual(w.value, 6.0)
+
+        #test creation with vardicts and kwvars
+        w = DepVar({'t':t, 'u':u}, v=v)
+        self.assertCountEqual(w.params, ('t','u','v'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v})
+        self.assertAlmostEqual(w.value, 6.0)
+
+    def test_dependency_graph(self):
+        """Test dependency_graph method."""
+        t = relentless.DesignVariable(value=1.0)
+        u = relentless.DesignVariable(value=2.0)
+        v = relentless.DesignVariable(value=3.0)
+
+        #test 1 level of dependency
+        w = DepVar(t=t, u=u, v=v)
+        g = w.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
+                              [['t'],['u'],['v']])
+
+        #test 2 levels of dependency
+        x = DepVar(t=t, v=v)
+        y = DepVar(w=w, x=x)
+        g = y.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w,x,y])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v),(x,t),(x,v),(y,w),(y,x)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
+                              [['t'],['u'],['v'],['t'],['v'],['w'],['x']])
+
+        #test more complex dependency
+        z = DepVar(t=t, w=w, y=y)
+        g = z.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w,x,y,z])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v),(x,t),(x,v),(y,w),(y,x),
+                                        (z,t),(z,w),(z,y)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
```

Comment:
At this point, you are probably convinced that the params are getting set, so you can just remove this check.